### ISSUE TITLE
Big integer 2i indexes sometimes sort incorrectly

### ISSUE THREAD

**Issue by reiddraper** [basho/riak_kv#499]
[As reported on the mailing list](http://lists.basho.com/pipermail/riak-users_lists.basho.com/2013-March/011276.html), sometimes bigger integers don't sort properly. This bug affects Riak 1.x. It only affects the leveldb backend, memory backend is _not_ affected. One manifestation of the issue can be reproduced like this:

```
curl -X PUT -H 'x-riak-index-foo_int:1362400142028' -d foo localhost:8098/buckets/new/keys/bar

# properly returns the 'bar' key
curl http://localhost:8098/buckets/new/index/foo_int/1100000000000/1362400142028

# does _not_ return the 'bar' key
curl http://localhost:8098/buckets/new/index/foo_int/1000000000000/1362400142028
```

We're pretty sure the issue has been tracked down to `sext`, specifically uwiger/sext#5.

**Comment by reiddraper** [comment]
Confirmed @uwiger's fix in sext also fixes the 2i issue.

**Comment by uwiger** [comment]
Are you happy with the migration support? The change in encoding may well make the problem worse while there are mixed encodings in the data set.

**Comment by reiddraper** [comment]
> Are you happy with the migration support?

We're still working through what the migration story is going to be. I think `sext:legacy_encode_bignum/1` will prove helpful though.

**Comment by uwiger** [comment]
Ok, let me know if you have any other ideas. Thanks for pinpointing the bug, and apologies for it being there in the first place.

**Comment by reiddraper** [comment]
Reminder for later, the fix is on `uwiger/sext`, not `esl/sext` yet, so we may just change the rebar.config if it doesn't make it over.

**Comment by uwiger** [comment]
It should be there now.

I haven't been pushing to esl in a while, but can start doing so again.

BR,
Ulf

2013/3/6 Reid Draper notifications@github.com

**Comment by jrwest** [comment]
addressed by #504. closing
